### ISSUE TITLE
Ask if user wants to create empty commit when there are no staged changes

### ISSUE THREAD

**Issue by jesseduffield** [jesseduffield/lazygit#2439]
**Is your feature request related to a problem? Please describe.**
I've got a custom command configured for adding an empty commit. Empty commits have a few uses. One of them is to create a conventional commit for the sake of an automated version bump.

Currently, when you press 'c' in the files panel and there are no staged changes, you get an error popup saying that there are no staged changes. Instead, I want to use a confirmation prompt, telling the user 'There are no staged changes. Would you like to create an empty commit?' where if the user says yes, it brings up the commit message panel as usual and upon pressing enter we create an empty commit with the given commit message.

**Describe alternatives you've considered**
The two obvious alternatives are:
* using a custom command
* having a built-in keybinding for creating an empty commit

The built-in keybinding approach does give you a little more control i.e. for when you actually do have staged changes but you want to create an empty commit anyway. Perhaps down the line we can add such a keybinding but I think the above feature idea still makes sense, and is orthogonal.

**Comment by jesseduffield** [comment]
I've chucked a 'good first issue' label on this, but I would hold off the work until https://github.com/jesseduffield/lazygit/pull/2390 is merged.

**Comment by jesseduffield** [comment]
Hmm here's the code we currently have in pkg/gui/controllers/helpers/working_tree_helper.go
```go
if len(self.model.Files) == 0 {
  return self.c.ErrorMsg(self.c.Tr.NoFilesStagedTitle)
}

if !self.AnyStagedFiles() {
  return self.PromptToStageAllAndRetry(self.HandleCommitPress)
}
```

I didn't realise we actually already prompt if there are no staged files. So this now becomes trickier. This makes me think the separate keybinding approach might actually be better

**Comment by moha-gh** [comment]
> I didn't realise we actually already prompt if there are no staged files. So this now becomes trickier. This makes me think the separate keybinding approach might actually be better

Yeah. At least I rely on the current behaviour a lot. If I know that I can simply commit everything it's usually `c` --> `Enter` for me :) So I'd also prefer a new keybinding (as I'd agree that it is useful to have a way to create an empty commit).

**Comment by mark2185** [comment]
`c` brings a menu and the first option is `stage all and commit`, and the second one being `empty commit`?

So that we can remove the confirmation panel and `c<Enter>` still works as usual.
